nwbwidgets/image.py

```diff
@@ -1,94 +1,167 @@
+
+    def _update_time_slider(self, value):
+        path_ext_file = value["new"]
+        # Read first frame
+        self.external_file = path_ext_file
+        tmax = (
+                self.imageseries.starting_time
+                + get_frame_count(path_ext_file)/self.imageseries.rate
+        )
+        tmin = 0
+        self.time_slider.max = tmax
+        self.time_slider.min = tmin
+        self._set_figure_external(tmin, self.external_file, tmin)
+
+    def _set_figure_3d(self, frame_number):
+        import ipyvolume.pylab as p3
+
+        output = widgets.Output()
+        p3.figure()
+        p3.volshow(
+            self.imageseries.data[frame_number].transpose([1, 0, 2]),
+            tf=linear_transfer_function([0, 0, 0], max_opacity=0.3),
+        )
+        output.clear_output(wait=True)
+        self.figure = output
+        with output:
+            p3.show()
+
+    def _set_figure_2d(self, frame_number):
+        data = self.imageseries.data[frame_number].T
+        if self.figure is None:
+            self.figure = go.FigureWidget(data=dict(type="image", z=data))
         else:
-            self.time_window_controller = foreign_time_window_controller
-        self.set_controls(**kwargs)
+            self._add_fig_trace(data, frame_number)
 
-        # Make widget figure
-        self.set_out_fig()
+    def _set_figure_external(self, time, ext_file_path, starting_time):
+        frame_number = self.time_to_index(time, starting_time)
+        data = get_frame(ext_file_path, frame_number)
+        if self.figure is None:
+            self.figure = go.FigureWidget(data=dict(type="image", z=data))
+        else:
+            self._add_fig_trace(data, frame_number)
 
-        self.children = [self.out_fig, self.time_window_controller]
+    def _add_fig_trace(self, img_data: np.ndarray, index):
+        self.figure.data[0]["z"] = img_data
+        self.figure.layout.title = f"Frame no: {index}"
 
-    def time_to_index(self, time):
+    def time_to_index(self, time, starting_time=None):
+        starting_time = (
+            starting_time
+            if starting_time is not None
+            else self.imageseries.starting_time
+        )
         if self.imageseries.external_file and self.imageseries.rate:
-            return int((time - self.imageseries.starting_time) * self.imageseries.rate)
+            return int((time - starting_time) * self.imageseries.rate)
         else:
             return timeseries_time_to_ind(self.imageseries, time)
 
-    def set_controls(self, **kwargs):
-        self.controls.update(
-            timeseries=fixed(self.imageseries), time_window=self.time_window_controller
-        )
-        self.controls.update({key: widgets.fixed(val) for key, val in kwargs.items()})
+    def get_children(self, *widgets):
+        set_widgets = [wid for wid in widgets if wid is not None]
+        return [self.figure, self.time_slider, *set_widgets]
```

Comment:
a foreign time slider should not be included in the children

Reply:
In case there is a `foreign_time_slider`, then what is the best way to add another callback (update figure) ? The usual way of using `foreign_time_slider.observe(set_figure)` would overwrite the old callbacks. 

One possible solution would be to link the values of `foreign_time_slider` to the new `time_slider`: `widgets.jslink((foreign_time_slider,"value"),(time_slider,"value"))`  In effect, the time_slider would now be controlled by the foreign slider while keeping the functionality of being changed independently. 

Check https://github.com/NeurodataWithoutBorders/nwb-jupyter-widgets/pull/196/commits/ae8e9cc42181bc6a21ea2d7786180cfd27b4eb7c